File: Data Structures & Algorithms/max-water-container/submission-2.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        left, right = 0 , len(heights) - 1
        maximum = 0

        while left < right : 
            current = min(heights[left], heights[right]) * (right - left)
            maximum = max(current, maximum)
            if heights[left] > heights[right]:
                right -= 1
            else:
                left += 1
            
        return maximum


        

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.java
class Solution {
    public int maxProfit(int[] prices) {
        int profit = 0;
        int left = 0;
        int right = left + 1;

        while (right < prices.length) {
            if (prices[left] > prices[right]) {
                left = right;
                right = left + 1;
            }
            else {
                profit = Math.max(profit, prices[right] - prices[left]);
                right++;
            }
        }
        return profit;
    }
}
